src/Controller/Component/PrgComponent.php

```diff
@@ -46,9 +46,10 @@ public function conversion($redirect = true)
             return null;
         }
         if ($redirect) {
-            return $this->_registry->getController()->redirect(
-                $this->request->params['pass'] + ['?' => $this->request->data]
-            );
+            $parsedUrl = parse_url($this->request->here(false));
```

Comment:
I'm not 100% certain here, because `here()` does parse the `query` params inside the method, so it might be easier to set `Request::query` to the request data before calling `here()`. I would be interested in other people's thoughts on this approach.